### DISCUSSION TITLE
Currently, what are the main differences between Skyfield and PyEphem, in terms of functionality?

### DISCUSSION THREAD

**Discussion by hidp123** [skyfielders/python-skyfield#776] [Q&A]
Currently, what are the main differences between Skyfield and PyEphem, in terms of functionality?

**Comment by davidmikolas** [comment]
Quick and non-authoritative answer to highlight some differences, but these are about accuracy and you've asked about "functionality" so I think there will be better answers forthcoming.

From https://rhodesmill.org/pyephem/

>PyEphem generates positions using the 1980s techniques popularized in [Jean Meeus’s Astronomical Algorithms](https://www.willbell.com/math/mc1.htm), like the IAU 1980 model of Earth nutation and VSOP87 planetary theory. These make PyEphem faster and more compact than modern astronomy libraries, but limit its accuracy to around 1 arcsecond. This is often sufficient for most amateur astronomy, but users needing higher precision should investigate a more modern Python astronomy library like Skyfield or AstroPy.

Whereas Skyfield loads the far more accurate and rigorous [JPL Development Ephemerides](https://en.wikipedia.org/wiki/Jet_Propulsion_Laboratory_Development_Ephemeris) ([pronunciation guide](https://english.stackexchange.com/q/370655))

In addition (I believe) that the way time, relativity and the speed of light (observed positions of solar system objects) is handled is more accurate and up to date.

**Comment by Bernmeister** [comment]
For my use case, Skyfield is essentially, functionally equivalent to PyEphem.  I use PyEphem to compute rise/set/az/alt for moon, sun, planets, minor planets, comets, stars and satellites.  Whilst in Skyfield solar eclipses are yet to be implemented, PyEphem doesn't have solar nor lunar eclipses, so Skyfield is ahead on that count!

If you are wondering whether to embark on using Skyfield or PyEphem, I'd recommend Skyfield first and see how you go; all depends on your use case scenario(s).
